### PR TITLE
Increase PHPStan level

### PR DESCRIPTION
**Pull request by ausi**
~~Level 6 seems impossible because psalm fights the changes required for phpstan.~~ Fixed by adding `@phpstan-` and `@psalm-` type annotations.

But the level 5 changes can be used I think.

/cc @m-vo

### CODE REVIEW

**Comment by ausi** [comment]
Level 6 works now, but we should probaly revert 8bd84150946624ab8a2a99cc2a21bff19225e55d and find out what the issue is here.

**Comment by m-vo** [comment]
see https://github.com/phpstan/phpstan-phpunit/issues/77

**Comment by leofeyer** [comment]
We should maybe rethink if we really need both PhpStan and Psalm. 🙈

File: src/Image.php [hunk 1]
```diff
@@ -41,6 +41,10 @@ class Image extends AbstractImage
      */
     private $palette;
 
+    /**
+     * @phpstan-param MetadataBag<mixed> $metadata
+     * @psalm-param MetadataBag $metadata
```
Comment (leofeyer) [inline src/Image.php:46]: Why does Psalm need this? It is exactly the same information as in the method signature.

Comment (ausi) [reply src/Image.php:46]: Psalm parses the `@phpstan-param` too and is confused by the generic notation `<mixed>`. I was only able to make both tools succeed by specifing both `@phpstan-` and `@psalm-`. I think if both tags are present every tool only uses its own tag.

Comment (leofeyer) [reply src/Image.php:46]: Hm, I see.

**Comment by ausi** [comment]
I would merge this as it is now, and create a new PR that reverts the `0 + ` changes to merge once phpstan/phpstan-phpunit#77 was resolved.

**Comment by leofeyer** [comment]
I would prefer to wait until the PHPStan devs have signaled that they want to fix phpstan/phpstan-phpunit#77. I am not that big a fan of merging pointless workarounds into production code just to fix issues in some dev tools.

**Comment by ausi** [comment]
The workaround only affects the `ImageTest` class. Production code is clean ☺️

**Comment by leofeyer** [comment]
Ok, that's better than. Still a pointless workaround. 🙈

**Comment by ausi** [comment]
Pull request that reverts the workaround: #27

**Pull request merged**